### ISSUE TITLE
Fine grained presubmit sample tests

### ISSUE THREAD

**Issue by dizcology** [googleapis/synthtool#1038]
**Is your feature request related to a problem? Please describe.**

Some library+sample repos (such as [python-aiplatform](https://github.com/googleapis/python-aiplatform/tree/master/samples/snippets)) has many samples and sample tests, which are frequently updated even if the library itself has not changed.  Presubmit checks would run all the sample tests even if a PR modifies only a few samples.  Some sample tests take a long time to run often causing the presubmit checks to take up to an hour.

 **Describe the solution you'd like**

Ideally, PR presubmit sample test runs should run only the modified sample tests and the samples tests whose corresponding samples have been modified in the PR.

If a PR changes anything outside of the samples folder, all sample tests should run.  (This is already the current behavior.)

 **Describe alternatives you've considered**

Alternatively, the granularity could be at directory level:  PR presubmit sample test runs should run only sample tests in subdirectories that have been changed in the PR whenever all changes in the PR are in the top level samples folder.

**Comment by SurferJeffAtGoogle** [comment]
@dizcology, I agree, but my first instinct is to update python-aiplatform so it behaves as you describe.  What changes do you imagine in this repository?
